### ISSUE TITLE
[BUG] XTERRA TRX3500 no control from QZ App

### ISSUE THREAD

**Issue by iamdmf** [cagnulein/qdomyos-zwift#1835]
[debug-Tue_Nov_28_10_30_41_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13487020/debug-Tue_Nov_28_10_30_41_2023.log)
[debug-Tue_Nov_28_10_26_51_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13487021/debug-Tue_Nov_28_10_26_51_2023.log)
[debug-Tue_Nov_28_10_30_57_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13487022/debug-Tue_Nov_28_10_30_57_2023.log)
[debug-Tue_Nov_28_10_38_32_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13487023/debug-Tue_Nov_28_10_38_32_2023.log)
[debug-Tue_Nov_28_10_45_23_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13487024/debug-Tue_Nov_28_10_45_23_2023.log)
**Describe the bug**

**To Reproduce**

**Expected behavior**
The treadmill is never controlled by the QZ app. I’ve got as far as having the inclination triangle flashing on a meet up run but my Zwift avatar never moves. I’ve tried toggling with the virtual device settings and restarting the initial ride and then run meet up but at no point does QZ control the treadmill although QZ can see my treadmill.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: iOS 
 - Version: 17.1.1

**Smartphone (please complete the following information):**
 - Device: iPhone 15 Pro Max
 - OS: iOS 17.1.1
 - Zwift is running on my iPad Pro.
 

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**

**Comment by cagnulein** [comment]
#1819 #574 ok thanks, but please don't open a new ticket, let's use this until the end. thanks

**Comment by cagnulein** [comment]
@iamdmf ok checked, the main issue is that the treadmill is disconnecting just after a connection. there is some strange thing. ok so let's try to diagnose this. Do you have an android device to do the same test? Just to understand if it's a iOS issue? Let me know

**Comment by iamdmf** [comment]
I don't unfortunately. Can my iPad run Zwift and QZ at the same time perhaps as a work around?

**Comment by cagnulein** [comment]
nope it's not the case. i will try to do a mod and I will send you a new beta version to try


send me an email to roberto.viola83 at gmail.com indicating this ticket . Thanks

**Comment by iamdmf** [comment]
Thanks! Email sent.

**Comment by cagnulein** [comment]
Sent! Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 28 nov 2023 alle ore 13:09 iamdmf ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
I sent you an email from apple to a new qz version, check the spam also

**Comment by iamdmf** [comment]
I've got it. It weirdly went straight to my delete folder. I'll give it a test now. I'll report back. Thanks

**Comment by iamdmf** [comment]
Wow wow wow!! It works!! Amazing! I can't use the QZ app to change the speed of the treadmill but that's fine I can just use the physical buttons on the treadmill as I do now! The only thing that appears to be missing is the heart rate from my Apple Watch? Any ideas there? Usually it pairs through the companion app. Thanks, David

**Comment by iamdmf** [comment]
Yes, i tried that but no readings came through. Perhaps i need to open the watch app first before opening the iPhone app? I'll have a play around with it over the coming days. Also I'm assuming i can join an event if I time my meet up to end just before the event starts? Essentially going into ride mode, join the run meet up and then jump from that into an event? Hopefully that works anyway. I'm very happy regardless. Thanks! :)
    On Tuesday, 28 November 2023 at 13:58:18 GMT, Roberto Viola ***@***.***> wrote:  
 
 


Great! you have to use the qz app on the watch too. Open it, press start and you will see the HR on the QZ app and then on zwift too
Let me know!

**Comment by cagnulein** [comment]
> Yes, i tried that but no readings came through.

on the watch app or in qz?



> erhaps i need to open the watch app first before opening the iPhone app?

nope the opposite



> lso I'm assuming i can join an event if I time my meet up to end just before the event starts?

yes


> Essentially going into ride mode, join the run meet up and then jump from that into an event? Hopefully that works anyway. I'm very happy regardless. Thanks! :)

yes but in order to do so, you need to be able to control speed and/or incline from qz. if they don't work, it means that I have to handle your treadmill in a different way (different protocol).

Let me know

**Comment by iamdmf** [comment]
The watch app was opened after, whilst i was on the treadmill but no readings came through onto the QZ iPhone app, nor in Zwift.
The QZ app doesn't control the speed of the treadmill. I tried increasing the speed through the app and nothing happened. When i use the buttons on the treadmill to increase speed the speed increased on the QZ app and in Zwift.
Thanks,David
    On Tuesday, 28 November 2023 at 14:20:04 GMT, Roberto Viola ***@***.***> wrote:  
 
 



Yes, i tried that but no readings came through.


on the watch app or in qz?


erhaps i need to open the watch app first before opening the iPhone app?


nope the opposite


lso I'm assuming i can join an event if I time my meet up to end just before the event starts?


yes


Essentially going into ride mode, join the run meet up and then jump from that into an event? Hopefully that works anyway. I'm very happy regardless. Thanks! :)


yes but in order to do so, you need to be able to control speed and/or incline from qz. if they don't work, it means that I have to handle your treadmill in a different way (different protocol).

Let me know

**Comment by cagnulein** [comment]
but are you seeing the hr on the qz app on the watch?

Il giorno mar 28 nov 2023 alle 15:52 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
I've just looked my watch app now and in isolation i can see the HR. I'll go test again as i dont think i saw it last time.
    On Tuesday, 28 November 2023 at 14:56:09 GMT, Roberto Viola ***@***.***> wrote:  
 
 

but are you seeing the hr on the qz app on the watch?

Il giorno mar 28 nov 2023 alle 15:52 iamdmf ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
ok let me know, in the mean time i'm building a new test version with a
completely different protocol for your treadmill, finger crossed

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 28 nov 2023 alle ore 16:00 iamdmf ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
ok I sent a new build for a different protocol for your treadmill. For today I guess I will not build any new build :)

**Comment by iamdmf** [comment]
Back to the avatar not moving I’m afraid. The treadmill didn’t connect like it did previously in the first test flight app. I can’t control the treadmill speed with the app. Feels like the previous version worked better. Thanks,DavidOn 28 Nov 2023, at 3:15 pm, Roberto Viola ***@***.***> wrote:﻿
ok I sent a new build for a different protocol for your treadmill. For today I guess I will not build any new build :)

**Comment by cagnulein** [comment]
ok so i will try another one. it will be simpler if you had an android
device.

Il giorno mar 28 nov 2023 alle 16:29 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
Thanks Roberto. Sorry i don't have an android device. 
    On Tuesday, 28 November 2023 at 15:36:42 GMT, Roberto Viola ***@***.***> wrote:  
 
 

ok so i will try another one. it will be simpler if you had an android
device.

Il giorno mar 28 nov 2023 alle 16:29 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
Thanks,DavidOn 28 Nov 2023, at 3:44 pm, Roberto Viola ***@***.***> wrote:﻿
can you send me a debug log so I can see how the treadmill answers to my

request of the new version?



Il giorno mar 28 nov 2023 alle 16:41 iamdmf ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
I don't see it, you have to upload it from the GitHub interface directly. Otherwise I can't see it.

**Comment by iamdmf** [comment]
[debug-Tue_Nov_28_15_48_46_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13490311/debug-Tue_Nov_28_15_48_46_2023.log)

**Comment by iamdmf** [comment]
Sorry, i've now added to github. :)
    On Tuesday, 28 November 2023 at 15:52:01 GMT, Roberto Viola ***@***.***> wrote:  
 
 


I don't see it, you have to upload it from the GitHub interface directly. Otherwise I can't see it.

**Comment by iamdmf** [comment]
[debug-Tue_Nov_28_15_48_46_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13490658/debug-Tue_Nov_28_15_48_46_2023.log)

**Comment by cagnulein** [comment]
hi @iamdmf in this log i don't see a connection to the treadmill at all. Could you please generate a new one? or maybe you have it already, it's just another file
Let me know

**Comment by cagnulein** [comment]
ok thanks, we should at least see a connection. Let me know if the error
persists

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 29 nov 2023 alle ore 09:59 iamdmf ***@***.***>
ha scritto:

**Comment by iamdmf** [comment]
[debug-Wed_Nov_29_10_47_20_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13499512/debug-Wed_Nov_29_10_47_20_2023.log)

**Comment by iamdmf** [comment]
Hi Roberto, I've added another debug log into github. There's definitely no connection to the treadmill. All apps were closed. The treadmill was turned on and then the QZ app opened. it says it found the treadmill but there's no BLE indicator on the dashboard of the treadmill. Zwift can see my iphone running the app but my avatar doesnt move even if i use the treadmill start button.
Thanks,David
    On Wednesday, 29 November 2023 at 09:28:18 GMT, Roberto Viola ***@***.***> wrote:  
 
 

ok thanks, we should at least see a connection. Let me know if the error
persists

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 29 nov 2023 alle ore 09:59 iamdmf ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
@iamdmf thanks. Log checked, the problem is that the QZ doesn't find the treadmill anymore. Are you able to see the treadmill from nrfconnect clearing the cache of it? It seems that the treadmill is not advertising itself anymore. Could you please double check it?

**Comment by iamdmf** [comment]
Thanks,DavidOn 29 Nov 2023, at 10:53 am, Roberto Viola ***@***.***> wrote:﻿
@iamdmf thanks. Log checked, the problem is that the QZ doesn't find the treadmill anymore. Are you able to see the treadmill from nrfconnect clearing the cache of it? It seems that the treadmill is not advertising itself anymore. Could you please double check it?

**Comment by iamdmf** [comment]
Hi Roberto, i sent you the screenshots of the nrf connect app. It connects to my treadmill. Do they not show the data you need? Thanks
    On Wednesday, 29 November 2023 at 12:41:11 GMT, Roberto Viola ***@***.***> wrote:  
 
 


sorry, i don't see the answers to my questions, am I missing something?

**Comment by cagnulein** [comment]
you have always to attach attachments to the github interface, otherwise I can't see them. thanks

**Comment by iamdmf** [comment]
![IMG_0421](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/0043060c-38e2-4db7-a99f-2c0f6f5bc70b)
![IMG_0422](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/2eac2208-a41c-4d4f-ab0d-e2b212340215)
![IMG_0423](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/a158ea03-0e60-4fd0-8e7c-342efb15efac)
![IMG_0424](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/882109cf-082e-4016-bd78-d4eabd5184f3)
![IMG_0425](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/6b5d972d-6b9b-40d6-b931-27bcac40f608)
![IMG_0426](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/b171eb22-5d29-41a3-b2ff-ce98c90b62ae)
![IMG_0427](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/bef81a50-eba9-4e38-a1fa-d812a29d7588)
![IMG_0428](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/90a932af-701f-4ed7-80b5-bdc763c4acd5)

**Comment by cagnulein** [comment]
so that's strange! could you please on the testflight app, going back to the previous version and check that it still works?

**Comment by iamdmf** [comment]
Ah apologies, I thought that was just for the debug logs. I’ve now added the screenshots to GitHub.Thanks,DavidOn 29 Nov 2023, at 12:46 pm, Roberto Viola ***@***.***> wrote:﻿
you have always to attach attachments to the github interface, otherwise I can't see them. thanks

**Comment by iamdmf** [comment]
[debug-Wed_Nov_29_13_26_41_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13501315/debug-Wed_Nov_29_13_26_41_2023.log)

**Comment by cagnulein** [comment]
ok so the previous one is working, let me check again the new code, i will be back soon

**Comment by iamdmf** [comment]
![IMG_0430](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/97bacb7b-bb68-4123-a781-df6b3386db05)

here you can see the BLE indicator:)

**Comment by cagnulein** [comment]
can you now download again the latest one and test? we have to see at least the tiles

**Comment by iamdmf** [comment]
![IMG_0431](https://github.com/cagnulein/qdomyos-zwift/assets/152033245/ce2ae744-b1bc-41a7-8d4d-804e0a1dac76)
[debug-Wed_Nov_29_14_03_55_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13501406/debug-Wed_Nov_29_14_03_55_2023.log)

**Comment by iamdmf** [comment]
Same as before, it doesnt connect nor does my avatar move. Picture and log uploaded to github.
Thanks
    On Wednesday, 29 November 2023 at 14:00:32 GMT, Roberto Viola ***@***.***> wrote:  
 
 


can you now download again the latest one and test? we have to see at least the tiles

**Comment by cagnulein** [comment]
ok now I can see the device finally, so yesterday the treadmill was in a
strange state. Ok let me check the log now

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 29 nov 2023 alle ore 15:09 iamdmf ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
ok got it, apart of qz, do you know about any other apps that can connect to your treadmill and control the incline? for example the app of the manufacturer? if so, which is the name of the app?

**Comment by iamdmf** [comment]
There are the manufacturers apps called Xterra and Xterra+. I've downloaded them previously but they are pretty rubbish apps. I can test and report back if that will be helpful? Thanks
    On Wednesday, 29 November 2023 at 14:12:29 GMT, Roberto Viola ***@***.***> wrote:  
 
 


ok got it, apart of qz, do you know about any other apps that can connect to your treadmill and control the incline? for example the app of the manufacturer? if so, which is the name of the app?

**Comment by cagnulein** [comment]
yes I need to understand if your treadmill is able to change the
inclination by hardware via 3rd party app. it could be that your treadmill
doesn't have this possibility.
for example, did you try kinomap?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 29 nov 2023 alle ore 15:22 iamdmf ***@***.***>
ha scritto:

**Comment by iamdmf** [comment]
Neither Xterra app allow the app to control the treadmill. All they do is display the data from the treadmill on the app. I've never tried kinomap. 
Thanks,David
    On Wednesday, 29 November 2023 at 14:24:33 GMT, Roberto Viola ***@***.***> wrote:  
 
 

yes I need to understand if your treadmill is able to change the
inclination by hardware via 3rd party app. it could be that your treadmill
doesn't have this possibility.
for example, did you try kinomap?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 29 nov 2023 alle ore 15:22 iamdmf ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
Try it so, we need to understand this first

Il giorno mer 29 nov 2023 alle 15:48 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
I've tested Kinomap, it's quite fun although it controls the incline there's no option to control the speed. The speed is just done with the buttons on the treadmill same as version one of the flight test app.
Thanks,David
    On Wednesday, 29 November 2023 at 14:54:12 GMT, Roberto Viola ***@***.***> wrote:  
 
 

Try it so, we need to understand this first

Il giorno mer 29 nov 2023 alle 15:48 iamdmf ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
so inclination control works on kinomap? if so can you try to borrow an
android device for just 10 minutes in order to get the log from it?

Il giorno mer 29 nov 2023 alle 17:08 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
Kinomap auto controls the incline like on the first TestFlight app of QZ with Zwift but there are no controls to use on the app to manually change the incline. I don’t think any app can control the treadmill. Thanks,DavidOn 29 Nov 2023, at 5:13 pm, Roberto Viola ***@***.***> wrote:﻿
 so inclination control works on kinomap? if so can you try to borrow an

android device for just 10 minutes in order to get the log from it?



Il giorno mer 29 nov 2023 alle 17:08 iamdmf ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
so i misunderstood it: on the first testflight are you able to change
incline from qz?

Il giorno mer 29 nov 2023 alle 19:05 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
No, but it auto inclines from Zwift. :)Thanks,DavidOn 29 Nov 2023, at 6:37 pm, Roberto Viola ***@***.***> wrote:﻿
so i misunderstood it: on the first testflight are you able to change

incline from qz?



Il giorno mer 29 nov 2023 alle 19:05 iamdmf ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
hah ok so we reached the target! i didn't understand it! ok so i will
revert my last changes and i will push to the master version!

Il giorno mer 29 nov 2023 alle 20:07 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
Brilliant! Thanks so much for all your efforts! When will the master version be available to use please?Thanks,DavidOn 29 Nov 2023, at 7:30 pm, Roberto Viola ***@***.***> wrote:﻿
hah ok so we reached the target! i didn't understand it! ok so i will

revert my last changes and i will push to the master version!



Il giorno mer 29 nov 2023 alle 20:07 iamdmf ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
you can use the previous test flight, it's the same ;)

Il giorno mer 29 nov 2023 alle 21:06 iamdmf ***@***.***> ha
scritto:

**Comment by iamdmf** [comment]
Yes I will. Just wondering when it gets rolled out to the master app? Only reason I ask is I noticed an expiry date on the test app. :)Thanks,DavidOn 29 Nov 2023, at 8:28 pm, Roberto Viola ***@***.***> wrote:﻿
you can use the previous test flight, it's the same ;)



Il giorno mer 29 nov 2023 alle 21:06 iamdmf ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
don't worry about that, i'm releasing beta updates to everyone every months.
i'm doing a production release twice a year usually in order to preserve
stability to the users

Il giorno mer 29 nov 2023 alle 22:55 iamdmf ***@***.***> ha
scritto:
